### ISSUE TITLE
Use bump with init_if_needed

### ISSUE THREAD

**Issue by leemurus** [coral-xyz/anchor#2996]
Is it possible to implement logic to use bump optimization when account exists?

**Comment by acheroncrypto** [comment]
I don't think this is possible currently, but it would be great to support this behavior.

**Comment by jamie-osec** [comment]
Done in v2, not v1
